waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -52,6 +60,10 @@ private ReconnectingMetastoreClientInvocationHandler(
       this.name = name;
       this.maxRetries = maxRetries;
       this.base = base;
+      if (base != null)
+        this.useSasl = base.getHiveConfBool(HiveConf.ConfVars.METASTORE_USE_THRIFT_SASL);
```

Comment:
We could rename this method to `isSaslEnabled`, remove the argument and then we won't need to expose the fact that the conf is being used etc.